openslides_backend/action/mediafile/set_as_font.py

```diff
@@ -0,0 +1,38 @@
+from typing import Any, Dict
+
+from ...models.models import Mediafile
+from ...shared.exceptions import ActionException
+from ...shared.patterns import FullQualifiedId
+from ..default_schema import DefaultSchema
+from ..generics import UpdateAction
+from ..register import register_action
+
+
+@register_action("mediafile.set_as_font")
+class MediafileSetAsFontAction(UpdateAction):
+    """
+    Action to set a mediafile as font.
+    """
+
+    model = Mediafile()
+    schema = DefaultSchema(Mediafile()).get_update_schema(
+        additional_required_fields={"place": {"type": "string"}},
+    )
+
+    def update_instance(self, instance: Dict[str, Any]) -> Dict[str, Any]:
+        """
+        Checks is_directory and mimetype and sets font.
+        """
+        mediafile = self.database.get(
+            FullQualifiedId(self.model.collection, instance["id"]),
+            ["is_directory", "mimetype", "meeting_id"],
+        )
+        place = instance["place"]
+        meeting_id = mediafile["meeting_id"]
+        if mediafile.get("is_directory"):
+            raise ActionException("Cannot set a directory as font.")
+        if mediafile.get("mimetype") not in ["font/otf", "font/ttf"]:
```

Comment:
We should match this to the client. It currently accepts `['font/ttf', 'font/woff', 'application/font-woff', 'application/font-sfnt']`.

Reply:
Done in https://github.com/OpenSlides/openslides-backend/pull/260